### ISSUE TITLE
Resolution of inheritance

### ISSUE THREAD

**Issue by VisLab** [hed-standard/hed-specification#577]
In translating HED to an ontology the issue of inheritance must be resolved.  HED schema attributes map nicely to `DataProperty` or `ObjectProperty` depending on their ranges.  **However**: `DataProperty` and `ObjectProperty` are always inherited by their subclasses.

This has caused me to review the behavior of inheritance of attributes (relevant to #532, #486, and the addition of `hedId` #444).
The new `hedId` being added in 8.3.0 cannot be inherited by its children -- it must be an owl `AnnotationProperty`. 

Therefore I propose:
1. Replacing the `isInheritedProperty` with `annotationProperty`. Attributes with the `annotationProperty` are not inherited.  
2. Attributes without the `annotationProperty` are inherited (`AnnotationProperty` in owl, `annotationProperty` in HED).
3. Inheritance only applies to tags (but since other schema elements don't have children this assumption isn't required).

I have evaluated our existing tags (`recommended` and `required` have been removed in 8.3.0, but since they don't appear in HED versions >= 8.0.0, this doesn't influence anything).

| Attribute | On tags? |  Inherited?  |  Notes |  
| --------- | ---------- | ---------  | ---------- |  
| `allowedCharacter`  | no   |   no |   |    
| `conversionFactor` | no | no |    |    
| `defaultUnits`  | no | no |    |    
|  `deprecatedFrom` | yes | yes |    |    
|  `extensionAllowed`  | yes | yes |  |   
| `hedId` | **yes** | **no** |  Should be an `annotationProperty`  |  
| `inLibrary` | yes | yes |    |    
| `relatedTag` | yes | yes |    |    
| `requireChild`  | yes | no |  Used for `Def`, `Def-expand` and`Definition`  -- meant for reserved tags. Make `annotationProperty` |    
| `reserved`  |  yes  | yes |  Unlikely to be extended anyway |  
| `rooted`  | yes | no  |  This is mainly for assembling schemas - could be an `annotationProperty` |    
|  `SIUnit`  | no | no |  |  
|  `SIUnitModifier`  |   no | no |   |  
| `SIUnitSymbolModifier` | no |  no  |   |  
| `suggestedTag` | yes | yes |  |  
| `tagGroup`  | yes | yes |  |  
| `takesValue` | yes | yes |  Can only be for placeholders so irrelevant for inheritance `annotationProperty` |
| `topLevelTagGroup` | yes | yes |   |  
| `unique` | yes | yes  |   |  
| `unitClass` | yes | yes |  Can only be for placeholders so irrelevant for inheritance |   
| `unitPrefix` | no | no |   |  
| `unitSymbol` | no | no |   |  
| `valueClass` | yes | yes | Can only be for placeholders so irrelevant for inheritance  |   

@IanCa @happy5214 --- I know this will require revisions of the schema validator (python) and schema loading (both python and javascript) but I don't see any other way to resolve the inheritance issues cleanly. Comments?

**In summary, the new inheritance criteria is:** 

`(schema has isInherited and isInherited) or  (schema does not have inherited and tagDomain and not annotationProperty)`

**Comment by VisLab** [comment]
This has been modified starting with HED schema 8.3.0 which is now in prerelease.

The `isInherited` schema property has been replaced by `annotationProperty`.  If a schema attribute has the `annotationProperty` it is NOT inherited, otherwise it is.  In otherwords most items are inherited or inheritance is irrelevant because the entity will never have children (e.g., unit modifiers for `SIUnitModifier`).

**Comment by VisLab** [comment]
This has been resolved with the release of 8.3.0.  Schema attributes that have the annotationProperty property are not inherited. All others are inherited.
